Shopify/cli
# [Themes] - Refactor `pull` command to prepare for public themes API

### WHY are these changes introduced?

Fixes https://github.com/Shopify/develop-advanced-edits/issues/341
Fixes https://github.com/Shopify/develop-advanced-edits/issues/342

Allows users to call `pull` programatically by providing the flags as arguments

### WHAT is this pull request doing?
1) Creates a service function `pull` which allows users to provide command flags as args
2) Adds JSDoc to the public `pull` method args
2) Splits the CLI2 logic (this is kept in the command layer)
3) Adds some tests to the service layer

### How to test your changes?
- Hover over the call to `pull` in `packages/theme/src/cli/commands/theme/pull.ts` to see the Docs
- Call `shopify-dev theme pull` with and without the `--legacy` flag. Everything should work as before

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
